conan/internal/runner/docker.py

```diff
@@ -64,13 +64,28 @@ def __init__(self, conan_api, command, host_profile, build_profile, args, raw_ar
         import docker
         import docker.api.build
         try:
-            self.docker_client = docker.from_env()
-            self.docker_api = docker.APIClient()
+            try:
+                self.docker_client = docker.from_env()
+                self.docker_api = docker.APIClient()
+            except:
+                docker_base_urls = [
+                    os.environ.get('CONAN_RUNNER_DOCKER_HOST'),
+                    os.environ.get('DOCKER_HOST'),
+                    'unix:///var/run/docker.sock',
+                    f'unix://{os.path.expanduser("~")}/.rd/docker.sock'
+                ]
+                for base_url in docker_base_urls:
+                    try:
+                        self.docker_client = docker.DockerClient(base_url=base_url, version='auto')
+                        break
+                    except:
+                        continue
```

Comment:
If every try fails, we could have a pretty error for the user in that case

Reply:
In that case, we rise this exception:

```python
raise ConanException("Docker Client failed to initialize."
                     "\n - Check if docker is installed and running"
                     "\n - Run 'pip install conan[runners]'")
```